sdks/python/apache_beam/coders/row_coder.py

```diff
@@ -82,8 +86,19 @@ def from_runner_api_parameter(schema, components, unused_context):
     return RowCoder(schema)
 
   @staticmethod
-  def from_type_hint(named_tuple_type, registry):
-    return RowCoder(named_tuple_to_schema(named_tuple_type))
+  def from_type_hint(type_hint, registry):
+    if isinstance(type_hint, row_type.RowTypeConstraint):
+      schema = schema_pb2.Schema(
+          fields=[
+              schema_pb2.Field(
+                  name=name,
+                  type=typing_to_runner_api(type))
+              for (name, type) in type_hint._fields
+          ],
+          id=str(uuid.uuid4()))
```

Comment:
Could you move this inference to `typehints.schemas` alongside `named_tuple_to_schema`? I have a WIP PR for batching schema'd PCollections that are inputs to Dataframes and I should re-use this logic there.

Reply:
Done.